### PR TITLE
3 get available payment methods

### PR DESCRIPTION
**Pull request by lojzatran**
Fixes: https://github.com/commercetools/commercetools-adyen-integration/issues/3

### CODE REVIEW

File: extension/src/ctp/ctp.js [hunk 1]
```diff
@@ -66,9 +72,19 @@ function setUpClient () {
...
     },
+
     fetch (uri) {
       return ctpClient.execute(this.buildRequestOptions(uri.build()))
     },
+
+    fetchBatches (uri, cb, opts = { accumulate: false }) {
+      return this.process(
+        this.buildRequestOptions(uri.build()),
+        data => Promise.resolve(cb(data.body.results)),
```
Comment (LEQADA) [inline extension/src/ctp/ctp.js]: `Promise.resolve()` is a native function. No need for bluebird

File: extension/src/ctp/ctp.js [hunk 2]
```diff
@@ -66,9 +72,19 @@ function setUpClient () {
         this.buildRequestOptions(uri.byId(id).build(), 'POST', body)
       )
     },
+
     fetch (uri) {
       return ctpClient.execute(this.buildRequestOptions(uri.build()))
     },
+
+    fetchBatches (uri, cb, opts = { accumulate: false }) {
```
Comment (LEQADA) [inline extension/src/ctp/ctp.js]: what is `cb`?

File: extension/src/api/payment/payment.controller.js [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
+  const handler = _getPaymentHandler(paymentObject)
+
+  const handlerResponse = await handler.handlePayment(paymentObject)
+  return httpUtils.sendResponse(response, undefined, undefined, handlerResponse)
 }
 
 async function _getPaymentObject (request) {
-  return httpUtils.collectRequestData(request)
+  const requestBody = JSON.parse(await httpUtils.collectRequestData(request))
+  return requestBody.resource.obj
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply extension/src/api/payment/payment.controller.js]: Yes.

File: extension/src/api/payment/payment.controller.js [hunk 4]
```diff
@@ -1,17 +1,27 @@
...
 }
 
 function _getPaymentHandler (paymentObject) {
+  for (const paymentHandler of paymentHandlers)
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply extension/src/api/payment/payment.controller.js]: Will be done here: https://github.com/commercetools/commercetools-adyen-integration/issues/14

File: extension/src/paymentHandler/fetchPaymentMethod.handler.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+const fetch = require('node-fetch')
+const configLoader = require('../config/config')
+const c = require('../config/constants')
+
+const config = configLoader.load()
+
+function isSupported (paymentObject) {
+  return paymentObject.paymentMethodInfo.paymentInterface === 'ctp-adyen-integration'
+    && !paymentObject.paymentMethodInfo.method
+    && paymentObject.custom.fields.countryCode
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply extension/src/paymentHandler/fetchPaymentMethod.handler.js]: Good point, I will do it as part of this task: https://github.com/commercetools/commercetools-adyen-integration/issues/14

File: extension/src/ctp/ctp.js (hunk 2 above)
Comment (lojzatran) [reply extension/src/ctp/ctp.js]: callback

File: extension/src/ctp/ctp.js (hunk 1 above)
Comment (lojzatran) [reply extension/src/ctp/ctp.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/5/commits/f37d84606eaf6a2ee4fdff3ef77b3ff8767f5e5c

File: extension/src/ctp/ctp.js (hunk 2 above)
Comment (lojzatran) [reply extension/src/ctp/ctp.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/5/commits/e2695da09020b78e0188bb10bdc741bfa88aa4f4

**Pull request merged**